CDS/scripts/js_utils/media-atom/hmac.js

```diff
@@ -0,0 +1,63 @@
+const crypto = require('crypto');
+const reqwest = require('reqwest');
+
+class HMACRequest {
+    constructor (configObj) {
+        const requiredConfig = ['shared_secret'];
+
+        requiredConfig.forEach(c => {
+            if (! Object.keys(configObj.config).includes(c)) {
+                throw `Invalid Config. Missing ${c}`;
+            }
+        });
+
+        this.configObj = configObj;
+
+        this.sharedSecret = this.configObj.config.shared_secret;
```

Comment:
Yep. The `Config` class follows a validation pattern which [catches missing properties](https://github.com/guardian/content_delivery_system/blob/aa-lint/CDS/scripts/js_utils/datastore/config.js#L62) upstream.

This does mean usage of `Config` looks a bit verbose however:

```js
const config = new Config();
config.validate()
  .then(() => {
    // do stuff
  })
  .catch(e => {
    // oh noes!
  });
```

Was tempted to do the validation in the constructor, but didn't like returning a promise in the constructor.

Happy to do this a cleaner way.